frontend/src/layout/navigation/navigationLogic.ts

```diff
@@ -65,6 +67,12 @@ export const navigationLogic = kea<navigationLogicType<UserType, SystemStatus, W
                 setHotkeyNavigationEngaged: (_, { hotkeyNavigationEngaged }) => hotkeyNavigationEngaged,
             },
         ],
+        filteredEnvironment: [
+            ENVIRONMENTS.PRODUCTION,
+            {
+                setFilteredEnvironment: (_, payload) => payload.environment,
```

Comment:
Nit: You can destructure the payload here to follow convention: `setFilteredEnvironment: (_, { environment } ) => environment`